compiler/src/dotty/tools/dotc/printing/PlainPrinter.scala

```diff
@@ -247,7 +247,10 @@ class PlainPrinter(_ctx: Context) extends Printer {
   }.close
 
   def toTextSingleton(tp: SingletonType): Text =
-    "(" ~ toTextRef(tp) ~ " : " ~ toTextGlobal(tp.underlying) ~ ")"
+    tp match {
+      case ConstantType(value) => toText(value)
```

Comment:
This will not work well with `Byte` and `Short`. We would not be able to know if `4` is a `Byte`, `Short`, `Int`.

Maybe we should keep `4: Byte` and `4: Short` as they are even if we cannot write them in source.

Reply:
Or maybe use `(4/*Byte*/)` and `(4/*Short*/)` to make it correct source code.

Reply:
Agreed. See my comment below